src/main/java/com/oneops/api/resource/Transition.java

```diff
@@ -373,7 +374,7 @@ public Release discardDeploymentPlan(String environmentName) throws OneOpsClient
 				if(response.getStatusCode() == 200 || response.getStatusCode() == 302) {
 					return response.getBody().as(Release.class);
 				} else {
-					String msg = String.format("Failed to discard deployment plan for environment %s due to %s", environmentName, response.getStatusLine());
+					String msg = String.format("Failed to discard deployment plan for environment %s. Status code: %s. Error: %s", environmentName, response.getStatusCode(), response.getBody().prettyPrint());
```

Comment:
`response.getBody().prettyPrint()` - Doesnt it cause NPE if body is `null`, because that's possible when the status code is non 2xx or 3xx.

Reply:
Good catch. Yeah, looks like I missed out adding checks in this method. 

In other methods where I want the error printed out, I am calling this method to check  if the response body is null - https://github.com/oneops/OneOpsAPIClient/blob/6d96b0732160c4b89f484f56359f3a379311583c/src/main/java/com/oneops/api/resource/Transition.java#L1893-L1901

An example call in the `deploy` method: https://github.com/oneops/OneOpsAPIClient/blob/6d96b0732160c4b89f484f56359f3a379311583c/src/main/java/com/oneops/api/resource/Transition.java#L281

I will make changes to this method as well.